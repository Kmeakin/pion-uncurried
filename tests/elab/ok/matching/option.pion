enum Option(T: Type) {
    None,
    Some(x: T),
}

let unwrap_or = fn(A: Type, it: Option(A), default: A) => match it {
    None => default,
    Some(y) => y,
};