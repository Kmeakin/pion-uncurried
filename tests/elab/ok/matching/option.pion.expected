enum Option(T: Type): Type {
    None: Option(T),
    Some(x: T): Option(T),
}

let unwrap_or: fn(A: Type, it: Option(A), default: A) -> A = fn(A: Type, it: Option(A), default: A) => match it {
    None => default,
    Some(y) => y,
};
